core/drivers/clk/clk-stm32mp13.c

```diff
@@ -1670,6 +1670,905 @@ static int stm32_clk_parse_fdt(const void *fdt, int node,
+	[MCE]		= &ck_mce,
+	[TIM2_K]	= &ck_tim2_k,
+	[TIM3_K]	= &ck_tim3_k,
+	[TIM4_K]	= &ck_tim4_k,
+	[TIM5_K]	= &ck_tim5_k,
+	[TIM6_K]	= &ck_tim6_k,
+	[TIM7_K]	= &ck_tim7_k,
+	[TIM13_K]	= &ck_tim13_k,
+	[TIM14_K]	= &ck_tim14_k,
+	[TIM1_K]	= &ck_tim1_k,
+	[TIM8_K]	= &ck_tim8_k,
+	[TIM16_K]	= &ck_tim16_k,
+	[TIM17_K]	= &ck_tim17_k,
+	[LTDC_PX]	= &ck_ltdc_px,
+	[DMA1]		= &ck_dma1,
+	[DMA2]		= &ck_dma2,
+	[MDMA]		= &ck_mdma,
+	[ETH1MAC]	= &ck_eth1mac,
+	[USBH]		= &ck_usbh,
+	[VREF]		= &ck_vref,
+	[TMPSENS]	= &ck_tmpsens,
+	[PMBCTRL]	= &ck_pmbctrl,
+	[HDP]		= &ck_hdp,
+	[STGENRO]	= &ck_stgenro,
+	[DMAMUX1]	= &ck_dmamux1,
+	[DMAMUX2]	= &ck_dmamux2,
+	[DMA3]		= &ck_dma3,
+	[TSC]		= &ck_tsc,
+	[AXIMC]		= &ck_aximc,
+	[CRC1]		= &ck_crc1,
+	[ETH1TX]	= &ck_eth1tx,
+	[ETH1RX]	= &ck_eth1rx,
+	[ETH2TX]	= &ck_eth2tx,
+	[ETH2RX]	= &ck_eth2rx,
+	[ETH2MAC]	= &ck_eth2mac,
+	[USART3_K]	= &ck_usart3_k,
+	[UART4_K]	= &ck_uart4_k,
+	[UART5_K]	= &ck_uart5_k,
+	[UART7_K]	= &ck_uart7_k,
+	[UART8_K]	= &ck_uart8_k,
+	[USART6_K]	= &ck_usart6_k,
+	[FMC_K]		= &ck_fmc_k,
+	[QSPI_K]	= &ck_qspi_k,
+	[LPTIM1_K]	= &ck_lptim1_k,
+	[SPI2_K]	= &ck_spi2_k,
+	[SPI3_K]	= &ck_spi3_k,
+	[SPDIF_K]	= &ck_spdif_k,
+	[SPI1_K]	= &ck_spi1_k,
+	[SPI4_K]	= &ck_spi4_k,
+	[SPI5_K]	= &ck_spi5_k,
+	[SAI1_K]	= &ck_sai1_k,
+	[SAI2_K]	= &ck_sai2_k,
+	[DFSDM_K]	= &ck_dfsdm_k,
+	[FDCAN_K]	= &ck_fdcan_k,
+	[I2C1_K]	= &ck_i2c1_k,
+	[I2C2_K]	= &ck_i2c2_k,
+	[ADFSDM_K]	= &ck_adfsdm_k,
+	[LPTIM2_K]	= &ck_lptim2_k,
+	[LPTIM3_K]	= &ck_lptim3_k,
+	[LPTIM4_K]	= &ck_lptim4_k,
+	[LPTIM5_K]	= &ck_lptim5_k,
+	[I2C3_K]	= &ck_i2c3_k,
+	[I2C5_K]	= &ck_i2c5_k,
+	[DCMIPP_K]	= &ck_dcmipp_k,
+	[ADC1_K]	= &ck_adc1_k,
+	[ADC2_K]	= &ck_adc2_k,
+	[ETH1CK_K]	= &ck_eth1ck_k,
+	[ETH2CK_K]	= &ck_eth2ck_k,
+	[CK_MCO1]	= &ck_mco1,
+	[CK_MCO2]	= &ck_mco2,
+	[CK_TRACE]	= &ck_trace,
+	[CK_OFF]	= &ck_off,
+	[USB_PHY_48]	= &ck_usb_phy_48Mhz,
+	[I2S_CKIN]	= &ck_i2sckin,
+};
+
+static bool clk_stm32_clock_is_critical(__maybe_unused struct clk *clk)
+{
+	struct clk *clk_criticals[] = {
+		&ck_hsi,
+		&ck_hse,
+		&ck_csi,
+		&ck_lsi,
+		&ck_lse,
+		&ck_pll2r,
+		&ck_mpu,
+		&ck_ddrc1,
+		&ck_ddrc1lp,
+		&ck_ddrphyc,
+		&ck_ddrphyclp,
+		&ck_ddrcapb,
+		&ck_ddrcapblp,
+		&ck_axidcg,
+		&ck_ddrphycapb,
+		&ck_ddrphycapblp,
+		&ck_rtcapb,
+		&ck_tzc,
+		&ck_etzpcb,
+		&ck_iwdg1apb,
+		&ck_bsec,
+		&ck_stgen_k,
+		&ck_bkpsram,
+		&ck_mce,
+		&ck_mco1,
+		&ck_rng1_k
+	};
+	size_t i = 0;
+
+	for (i = 0; i < ARRAY_SIZE(clk_criticals); i++) {
+		struct clk *clk_critical = clk_criticals[i];
+
+		if (clk == clk_critical)
+			return true;
+	}
+
+	return false;
+}
+
+static void clk_stm32_init_oscillators(const void *fdt, int node)
+{
+	size_t i = 0;
+	const char *name[6] = { "clk-hse", "clk-hsi", "clk-lse",
+				"clk-lsi", "clk-csi", "clk-i2sin" };
+	struct clk *clks[6] = { &ck_hse, &ck_hsi, &ck_lse,
+				&ck_lsi, &ck_csi, &ck_i2sckin };
+
+	for (i = 0; i < ARRAY_SIZE(clks); i++) {
+		struct clk *clk = NULL;
+
+		clk_dt_get_by_name(fdt, node, name[i], &clk);
```

Comment:
check return value

Reply:
don't need, if the oscillator is not found i want to fill my table with NULL (clks[i]->parents[0] = clk)

Reply:
ok